src/hal/i_components/knob2float.icomp

```diff
@@ -34,8 +34,8 @@ license "GPL";
 
 FUNCTION(_)
 {
-    long delta_counts;
-    double tmp_out;
+    hal_float_t delta_counts;
```

Comment:
OK, you are saying that the types they should be would need a typecast, 
I was saying the ones they currently are do not.

Changed to s32 and typecast accordingly